packages/manager/apps/web/client/app/hosting/automated-emails/hosting-automated-emails.controller.js

```diff
@@ -150,7 +68,144 @@ export default class HostingTabAutomatedEmailsCtrl {
       this.guide = get(guides, 'hostingScriptEmail', null);
     });
 
-    this.retrievingAutomatedEmails();
+    return this.retrievingAutomatedEmails();
+  }
+
+  loadChart() {
+    this.loaders.volumes = true;
+    this.HostingStatistics.getMetricsToken(this.$stateParams.productId).then(
+      ({ endpoint, token }) => {
+        const query = `sum without(cluster, statusCode,cluster_name, datacenter, host, host_type, hw_profile, service_name, user) (label_replace(sum_over_time((mailout_sendmails_count{service_name="${this.$stateParams.productId}"}[30m])), "mail", "sent", "", "")) OR label_replace(vector(0), "mail", "sent", "", "")`;
+
+        this.stats.chart = new this.ChartFactory({
+          type: 'line',
+          plugins: [ChartDatasourcePrometheusPlugin],
+          options: {
+            responsive: true,
+            maintainAspectRatio: false,
+            plugins: {
+              'datasource-prometheus': {
+                borderWidth: 1,
+                errorMsg: {
+                  message: this.$translate.instant(
+                    'hosting_tab_STATISTICS_none',
+                  ),
+                },
+                noDataMsg: {
+                  message: this.$translate.instant(
+                    'hosting_tab_STATISTICS_none',
+                  ),
+                },
+                findInBorderColorMap: () => '#848CBC',
+                findInBackgroundColorMap: () => '#EAECF4',
+                findInLabelMap: () =>
+                  this.$translate.instant(
+                    'hosting_tab_AUTOMATED_EMAILS_emails_sent',
+                  ),
+                fill: true,
+                tension: 0.5,
+                query: (start, end) => {
+                  const url = `${endpoint}/prometheus/api/v1/query_range?query=${query}&start=${start.toISOString()}&end=${end.toISOString()}&step=30m`;
+
+                  const headers = {
+                    authorization: `bearer ${token}`,
+                    'content-type': 'application/x-www-form-urlencoded',
+                  };
+
+                  return fetch(url, { headers })
+                    .then((response) => {
+                      if (response.ok) {
+                        return response.json();
+                      }
+
+                      return null;
+                    })
+                    .then((response) => response.data);
```

Comment:
Will should be `$http`

Reply:
No needs and avoid angularjs dependencies

Reply:
Yes I understand your argument but it's the practice for this app